In [1]:
import plotly_express as px
import pandas as pd

# Importing dataset from excel file

In [14]:
df_vaccine_data = pd.read_excel("./Data/folkhalsomyndigheten_Covid19_Vaccine.xlsx", sheet_name= "Vaccinerade kommun och ålder")

In [15]:
df_vaccine_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2900 entries, 0 to 2899
Data columns (total 14 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Län                  2900 non-null   int64  
 1   Län_namn             2900 non-null   object 
 2   Kommun               2900 non-null   int64  
 3   Kommun_namn          2900 non-null   object 
 4   Ålder                2900 non-null   object 
 5   Befolkning           2900 non-null   int64  
 6   Antal minst 1 dos    2900 non-null   int64  
 7   Antal minst 2 doser  2900 non-null   int64  
 8   Antal 3 doser        2320 non-null   float64
 9   Antal 4 doser        870 non-null    float64
 10  Andel minst 1 dos    2900 non-null   float64
 11  Andel minst 2 doser  2900 non-null   float64
 12  Andel 3 doser        2320 non-null   float64
 13  Andel 4 doser        870 non-null    float64
dtypes: float64(6), int64(5), object(3)
memory usage: 317.3+ KB


In [16]:
df_vaccine_data.head(7)

Län        Län_namn  Kommun     Kommun_namn  Ålder  Befolkning  \
0    1  Stockholms län     114  Upplands Väsby  12-15        2422   
1    1  Stockholms län     114  Upplands Väsby  16-17        1203   
2    1  Stockholms län     114  Upplands Väsby  18-29        6692   
3    1  Stockholms län     114  Upplands Väsby  30-39        7332   
4    1  Stockholms län     114  Upplands Väsby  40-49        6946   
5    1  Stockholms län     114  Upplands Väsby  50-59        6332   
6    1  Stockholms län     114  Upplands Väsby  60-69        4258   

   Antal minst 1 dos  Antal minst 2 doser  Antal 3 doser  Antal 4 doser  \
0               1206                 1046            NaN            NaN   
1                839                  755            NaN            NaN   
2               4887                 4469         1959.0            NaN   
3               5542                 5240         2878.0            NaN   
4               5592                 5429         3719.0            NaN   
5               5644                 5558         4535.0            NaN   
6               3829                 3786         3335.0            NaN   

   Andel minst 1 dos  Andel minst 2 doser  Andel 3 doser  Andel 4 doser  
0           0.497936             0.431874            NaN            NaN  
1           0.697423             0.627598            NaN            NaN  
2           0.730275             0.667812       0.292738            NaN  
3           0.755865             0.714675       0.392526            NaN  
4           0.805068             0.781601       0.535416            NaN  
5           0.891346             0.877764       0.716203            NaN  
6           0.899248             0.889150       0.783232            NaN

### Task A
Counties in the dataset

In [17]:
df_vaccine_data["Län_namn"].unique()

array(['Stockholms län', 'Uppsala län', 'Södermanlands län',
       'Östergötlands län', 'Jönköpings län', 'Kronobergs län',
       'Kalmar län', 'Gotlands län', 'Blekinge län', 'Skåne län',
       'Hallands län', 'Västra Götalands län', 'Värmlands län',
       'Örebro län', 'Västmanlands län', 'Dalarnas län', 'Gävleborgs län',
       'Västernorrlands län', 'Jämtlands län', 'Västerbottens län',
       'Norrbottens län'], dtype=object)

In [18]:
df_vaccine_data["Län_namn"].nunique()

21

##### There are 21 counties in sweden, so the dataset seems to have data about all of them.

### Task B
Municipalities in the dataset

In [22]:
df_vaccine_data["Kommun_namn"].unique()

array(['Upplands Väsby', 'Vallentuna', 'Österåker', 'Värmdö', 'Järfälla',
       'Ekerö', 'Huddinge', 'Botkyrka', 'Salem', 'Haninge', 'Tyresö',
       'Upplands-Bro', 'Nykvarn', 'Täby', 'Danderyd', 'Sollentuna',
       'Stockholm', 'Södertälje', 'Nacka', 'Sundbyberg', 'Solna',
       'Lidingö', 'Vaxholm', 'Norrtälje', 'Sigtuna', 'Nynäshamn', 'Håbo',
       'Älvkarleby', 'Knivsta', 'Heby', 'Tierp', 'Uppsala', 'Enköping',
       'Östhammar', 'Vingåker', 'Gnesta', 'Nyköping', 'Oxelösund', 'Flen',
       'Katrineholm', 'Eskilstuna', 'Strängnäs', 'Trosa', 'Ödeshög',
       'Ydre', 'Kinda', 'Boxholm', 'Åtvidaberg', 'Finspång',
       'Valdemarsvik', 'Linköping', 'Norrköping', 'Söderköping', 'Motala',
       'Vadstena', 'Mjölby', 'Aneby', 'Gnosjö', 'Mullsjö', 'Habo',
       'Gislaved', 'Vaggeryd', 'Jönköping', 'Nässjö', 'Värnamo', 'Sävsjö',
       'Vetlanda', 'Eksjö', 'Tranås', 'Uppvidinge', 'Lessebo', 'Tingsryd',
       'Alvesta', 'Älmhult', 'Markaryd', 'Växjö', 'Ljungby', 'Högsby',
       '

In [23]:
df_vaccine_data["Kommun_namn"].nunique()

290

##### There are 290 counties in sweden, so the dataset seems to have data about all of them aswell.